File: services/backend-api/Qrbank/src/main/java/backend_api/Qrbank/model/entities/Account.java
package backend_api.Qrbank.model.entities;

import backend_api.Qrbank.model.Model;
import backend_api.Qrbank.model.enums.IbanCurrency;
import jakarta.validation.constraints.NotBlank;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.relational.core.mapping.Column;
import org.springframework.data.relational.core.mapping.Table;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
@Table("accounts")
public class Account {

    @Id
    @NotBlank
    @Column("id")
    protected Long id;

    @NotBlank
    @Column("created_at")
    protected LocalDateTime createdAt;

    @NotBlank
    @Column("updated_at")
    protected LocalDateTime updatedAt;

    @NotBlank
    @Column("deleted_at")
    protected LocalDateTime deletedAt;
    @NotBlank
    @Column("user_id")
    private Long userId;

    @NotBlank
    @Column("account_number")
    private String accountNumber;

    @NotBlank
    @Column("iban")
    private String iban;

    @NotBlank
    @Column("balance")
    private BigDecimal balance;

    @NotBlank
    @Column("currency")
    private IbanCurrency currency;

    @NotBlank
    @Column("active")
    private boolean active;

}

File: app/mobile/qrbank_app/lib/model/account.dart
class Account {
  final String id;
  final String accountNumber;
  final String iban;
  final double balance;
  final String currency;
  final bool isActive;
  final DateTime createdAt;
  final DateTime? updatedAt;
  final DateTime? deletedAt;

  Account({
    required this.id,
    required this.accountNumber,
    required this.iban,
    required this.balance,
    required this.currency,
    required this.isActive,
    required this.createdAt,
    this.updatedAt,
    this.deletedAt,
  });

  factory Account.fromJson(Map<String, dynamic> json) {
    return Account(
      id: json['id'],
      accountNumber: json['accountNumber'],
      iban: json['iban'],
      balance: (json['balance'] as num).toDouble(),
      currency: json['currency'],
      isActive: json['isActive'],
      createdAt: DateTime.parse(json['createdAt']),
      updatedAt: DateTime.parse(json['updatedAt']),
      deletedAt:
          json['deletedAt'] != null ? DateTime.parse(json['deletedAt']) : null,
    );
  }
}